### ISSUE TITLE
`traits` don't appear on `context` on tracking events

### ISSUE THREAD

**Issue by Wiingaard** [segmentio/analytics-react-native#586]
After upgrading the library from version `1.5.0` to `2.3.1` I realized that user-identified traits no longer appear in the `context` of tracking events, now instead they appear as a `trait` object directly on the BaseEvent. I described this in more detail in my [original issue](https://github.com/segmentio/analytics-react-native/issues/578#issuecomment-1167081358) here. 

This is a problem for us because we rely on exporting data to multiple destinations in a specific format. Such as exporting `context_traits_first_name` to BigQuery. This will become `traits_first_name` after the update I assume. We also can't simply change the integration, because this will not be backward compatible with the old app versions still running the old SDK.

<img width="489" alt="Screenshot 2022-06-27 at 11 39 54" src="https://user-images.githubusercontent.com/8324312/176135959-dfd009ee-ffd7-49d7-9346-9b269c331242.png">

- `analytics-react-native` version: `2.3.1`
- Integrations versions (if used): `"@segment/analytics-react-native-plugin-amplitude-session": "0.2.1"`
- React Native version: `0.65.2`
- iOS or Android or both? Both

**Steps to reproduce**

1. Create an SDK client.
2. Identify the user
3. Fire a tracking event

Here's [source code](https://github.com/Wiingaard/DebugSegment/tree/master) that can reproduce the issue.

**Expected behavior**

When inspecting the tracking event in the online debugger, I expect the identified traits to be attached to the tracking event at the JSON path `context.traits`.

**Actual behavior**

Instead, the traits are added to as a base of the JSON, directly at path `traits`. This is a problem for us, because when we update, we'll have old versions of our app sending traits into the context of the event, and new versions of the app sending traits into the base traits object. This is messing with how data is exported to destinations.

**Comment by Wiingaard** [comment]
I just ran another new experiment:

1. I created a new tracking event called `'Some New Debug Event'` with properties `{'hello': true}`
2. I added a new user trait property called `{'wow': true}`
3. I open the app, verity that Identify is called and trigger the tracking event afterwards.
4. When Inspecting the online debugger, I see the event. `hello` appear in `properties.hello` (as expected), and `wow` appears in `traits.wow`. I expected `wow` to appear in `context.traits.wow`.
5. When inspecting the `BigQuery` destination for this source, I can find the new collection `some_new_debug_event` as expected, and I see the property `hello` under properties as expected, but the trait `wow` doesn't appear under properties like I would expect. Neither as `traits_wow` or `context_traits_now`. 

So the traits are not able to be synced with the destination. This is breaking the schema and preventing us from upgrading to v2 of this library, which is an issue due to the [non-compliant Android SDK-issue](https://github.com/segmentio/analytics-android/issues/802).

**Comment by trevorzablocki2** [comment]
@Wiingaard I just put up a PR that makes the traits compatible with how they worked in version 1.x I think.

**Comment by oscb** [comment]
I just checked with the team. This behavior was deprecated in our new libraries as some device mode destinations reject events with user identifiable data due to their terms of service. I just reverted the change in our last release not to remove them at the root level as well.

It is still possible to accomplish this with a custom plugin. I added an example [here](https://github.com/segmentio/analytics-react-native/blob/master/example/src/plugins/InjectTraits.ts). You would need to drop this file into your code and add it as a plugin to your client:

```ts
import { createClient } from '@segment/analytics-react-native';

import { InjectTraits } from './InjectTraits';

const segmentClient = createClient({
  writeKey: 'SEGMENT_KEY'
});

segmentClient.add({ plugin: new InjectTraits() });
```

Added the note to our migration guide: https://github.com/segmentio/analytics-react-native/blob/master/MIGRATION_GUIDE.md#context-traits

For the moment is not something we plan to release officially into NPM cause of this incompatibility but feel free to use this.

**Comment by Wiingaard** [comment]
Thank you again, Oscar 🏆

I've also been in contact with your support team, who also mentioned that traits have been removed in version 2. So I was a bit confused about the changes in version `2.3.1`. Thanks for reverting and providing a very helpful example and the documentation to the migration guide. 

I've updated to version `2.3.2` and implemented a plugin as you suggested. I've verified that traits now appear in the debugger and are available to export to BigQuery in the format that they used to. I did one slight modification to the plugin though, because in the example you're dumping the entire context back into the traits. Consider updating the example to something like this:
```
class InjectTraits extends PlatformPlugin {
    type = PluginType.before

    execute(event: SegmentEvent) {
        return {
            ...event,
            context: {
                ...event.context,
                traits: this.analytics?.userInfo.get().traits
            }
        }
    }
}
```
